Use double precision for CalcExpressionBlendLength.

mattwoodrow opened the pull request:
#### 53bb745eddb3c6459fa614930574c83244fdd891
<pre>
Use double precision for CalcExpressionBlendLength.
<a href="https://bugs.webkit.org/show_bug.cgi?id=249563">https://bugs.webkit.org/show_bug.cgi?id=249563</a>
&lt;rdar://103496037&gt;

Reviewed by Simon Fraser and Sam Weinig.

The incoming type from BlendingContext is a double, so we should keep it as double to improve the precision of the evaluated blend function.

This fixes the perspective-origin-interpolation.html WPT test on arm64, and transform-origin-interpolation everywhere.

* Source/WebCore/platform/calc/CalcExpressionBlendLength.cpp:
(WebCore::CalcExpressionBlendLength::CalcExpressionBlendLength):
(WebCore::CalcExpressionBlendLength::evaluate const):
* Source/WebCore/platform/calc/CalcExpressionBlendLength.h:

Canonical link: <a href="https://commits.webkit.org/258072@main">https://commits.webkit.org/258072@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f586b7a5a3850750de45b55db0c69267b3f5270e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100829 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9977 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33874 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110131 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170405 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104818 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10910 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/779 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93237 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107976 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106612 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8243 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91501 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34863 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90152 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22896 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/77838 "Found 2 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3669 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24416 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3688 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/745 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9799 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43916 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5537 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5466 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
Looks like there's another progression to land new results for.

mattwoodrow committed:
Use double precision for CalcExpressionBlendLength.
https://bugs.webkit.org/show_bug.cgi?id=249563
<rdar://103496037>

Reviewed by Simon Fraser and Sam Weinig.

The incoming type from BlendingContext is a double, so we should keep it as double to improve the precision of the evaluated blend function.

This fixes the perspective-origin-interpolation.html WPT test on arm64, and transform-origin-interpolation everywhere.

* Source/WebCore/platform/calc/CalcExpressionBlendLength.cpp:
(WebCore::CalcExpressionBlendLength::CalcExpressionBlendLength):
(WebCore::CalcExpressionBlendLength::evaluate const):
* Source/WebCore/platform/calc/CalcExpressionBlendLength.h:

Canonical link: https://commits.webkit.org/258072@main
